lib/std/math.zig

```diff
@@ -968,48 +922,11 @@ fn testRem() !void {
     try testing.expectError(error.DivisionByZero, rem(f32, 10, 0));
 }
 
-/// Returns the absolute value of a floating point number.
+/// Returns the absolute value of an integer or a floating point number.
 /// Uses a dedicated hardware instruction when available.
-/// This is the same as calling the builtin @fabs
-pub inline fn fabs(value: anytype) @TypeOf(value) {
-    return @fabs(value);
-}
-
-/// Returns the absolute value of the integer parameter.
-/// Converts result type to unsigned if needed and returns a value of an unsigned integer type.
-/// Use `absInt` if you want to keep your integer type signed.
-pub fn absCast(x: anytype) switch (@typeInfo(@TypeOf(x))) {
-    .ComptimeInt => comptime_int,
-    .Int => |int_info| std.meta.Int(.unsigned, int_info.bits),
-    else => @compileError("absCast only accepts integers"),
-} {
-    switch (@typeInfo(@TypeOf(x))) {
-        .ComptimeInt => {
-            if (x < 0) {
-                return -x;
-            } else {
-                return x;
-            }
-        },
-        .Int => |int_info| {
-            if (int_info.signedness == .unsigned) return x;
-            const Uint = std.meta.Int(.unsigned, int_info.bits);
-            if (x < 0) {
-                return ~@as(Uint, @bitCast(x +% -1));
-            } else {
-                return @as(Uint, @intCast(x));
-            }
-        },
-        else => unreachable,
-    }
-}
-
-test "absCast" {
-    try testing.expectEqual(@as(u1, 1), absCast(@as(i1, -1)));
-    try testing.expectEqual(@as(u32, 999), absCast(@as(i32, -999)));
-    try testing.expectEqual(@as(u32, 999), absCast(@as(i32, 999)));
-    try testing.expectEqual(@as(u32, -minInt(i32)), absCast(@as(i32, minInt(i32))));
-    try testing.expectEqual(999, absCast(-999));
+/// This is the same as calling the builtin @abs
+pub inline fn abs(value: anytype) @TypeOf(value) {
+    return @abs(value);
```

Comment:
this should be removed or replaced with a `@compileError("deprecated; use @abs");`

Reply:
I also noticed that `std.math.fabs/abs` isn't really used anywhere except in `lib/compiler_rt/divc3.zig` where we might as well use `@fabs/@abs`, I also just now noticed that the way it's implemented currently won't work because the return type of `@abs` will be different if the operand is a signed integer. I didn't remove it at first because there are a lot more aliases for the math builtins and they were added back after having been removed about 1 year ago (see #11666). I didn't feel like this was the place to remove just one alias but it does make sense to remove if it doesn't work.